libsolidity/analysis/TypeChecker.cpp

```diff
@@ -791,7 +793,10 @@ bool TypeChecker::visit(TupleExpression const& _tuple)
 			}
 			else
 				types.push_back(TypePointer());
-		_tuple.annotation().type = make_shared<TupleType>(types);
+		if (_tuple.isInlineArray())
```

Comment:
Please rather remove that, it is unreachable because of the assert above (it will throw) and should be addressed in the next PR anyway.